cirq-core/cirq/ops/classically_controlled_operation.py

```diff
@@ -68,13 +70,26 @@ def __init__(
             raise ValueError(
                 f'Cannot conditionally run operations with measurements: {sub_operation}'
             )
-        keys = tuple(value.MeasurementKey(c) if isinstance(c, str) else c for c in conditions)
         if isinstance(sub_operation, ClassicallyControlledOperation):
-            keys += sub_operation._control_keys
+            conditions += sub_operation._conditions
```

Comment:
This would update the input if the user passes in a List (ignoring type warnings). Looks like this is an existing bug too.